File: 3184-count-pairs-that-form-a-complete-day-i/3184-count-pairs-that-form-a-complete-day-i.py
class Solution:
    def countCompleteDayPairs(self, hours: List[int]) -> int:
        return sum(
            1 for i in range(len(hours)-1) 
            for j in range(i+1, len(hours)) 
            if (hours[i] +hours[j]) % 24 == 0
            )
        

File: 3184-count-pairs-that-form-a-complete-day-i/3184-count-pairs-that-form-a-complete-day-i.rb
# @param {Integer[]} hours
# @return {Integer}
def count_complete_day_pairs(hours)
    hours.each_index.sum do |i|
        (i+1...hours.size).count do |j|
            (hours[i] +hours[j]) % 24 == 0  
        end 
    end

end